search/search-client-query-transformer/src/main/java/io/camunda/search/clients/SearchClients.java

```diff
@@ -184,13 +188,58 @@ public SearchQueryResult<ProcessDefinitionEntity> searchProcessDefinitions(
   @Override
   public SearchQueryResult<ProcessInstanceEntity> searchProcessInstances(
       final ProcessInstanceQuery filter) {
+    if (!filter.filter().incidentErrorHashCodes().isEmpty()) {
+      return searchProcessInstancesByIncidentErrorHash(filter);
+    }
+
     return getSearchExecutor()
         .search(
             filter,
             io.camunda.webapps.schema.entities.operate.listview.ProcessInstanceForListViewEntity
                 .class);
   }
 
+  private SearchQueryResult<ProcessInstanceEntity> searchProcessInstancesByIncidentErrorHash(
+      final ProcessInstanceQuery filter) {
+
+    final var originalFilter = filter.filter();
+
+    final var incidentFilter =
+        FilterBuilders.incident(
+            f ->
+                f.errorMessageHashes(originalFilter.incidentErrorHashCodes())
+                    .states(IncidentState.ACTIVE));
+
+    final var incidentResult = searchIncidents(IncidentQuery.of(f -> f.filter(incidentFilter)));
+
+    if (incidentResult.items().isEmpty()) {
+      return new SearchQueryResult.Builder<ProcessInstanceEntity>().build();
+    }
+
+    final var processInstanceKeys =
+        incidentResult.items().stream().map(IncidentEntity::processInstanceKey).toList();
+
+    final var updatedFilter =
+        Builder.from(originalFilter)
+            .processInstanceKeyOperations(Operation.in(processInstanceKeys))
+            .hasIncident(true)
```

Comment:
💭 What if the original filter also contained processInstanceKeys? 😀

Kind of edge case but it's possible. I guess, we could add those operations too, if there is any.